fmlloader/src/main/java/net/minecraftforge/fml/loading/moddiscovery/AbstractModProvider.java

```diff
@@ -37,11 +37,11 @@ public abstract class AbstractModProvider implements IModProvider {
     protected static final String MODS_TOML = "META-INF/mods.toml";
 
     protected IModLocator.ModFileOrException createMod(Path path) {
-        return createMod(path, false);
+        return createMod(path, getDefaultJarModType(), false);
     }
 
     @Nullable
-    protected IModLocator.ModFileOrException createMod(Path path, boolean ignoreUnknown) {
+    protected IModLocator.ModFileOrException createMod(Path path, String defaultType, boolean ignoreUnknown) {
```

Comment:
binary break, you just need to add a bouncer method like the other. Also best to add the new parameter at the end of the list instead of in the middle.